cvat-ui/src/components/labels-editor/constructor-viewer-item.tsx

```diff
@@ -22,11 +22,26 @@ export default function ConstructorViewerItem(props: ConstructorViewerItemProps)
         color, label, onUpdate, onDelete,
     } = props;
 
+    const backgroundColor = color || consts.NEW_LABEL_COLOR;
+    let textColor = '#ffffff';
+    try {
+        // convert color to grayscale and from the result get better text color
+        // (for darken background -> lighter text, etc.)
+        const [r, g, b] = [backgroundColor.slice(1, 3), backgroundColor.slice(3, 5), backgroundColor.slice(5, 7)];
+        const grayscale = (parseInt(r, 16) + parseInt(g, 16) + parseInt(b, 16)) / 3;
+        if (grayscale - 128 >= 0) {
```

Comment:
What do you think about using something like HSL?
```
let [r, g, b] = [backgroundColor.slice(1, 3), backgroundColor.slice(3, 5), backgroundColor.slice(5, 7)];
r = parseInt(r, 16) / 255;
g = parseInt(g, 16) / 255;
b = parseInt(b, 16) / 255;
const CMax = Math.max(r, g, b);
const CMin = Math.min(r, g, b);
const lightness = (CMax + CMin) / 2;
if (lightness - 0.85 >= 0) {
      textColor = '#000000';
}
```
HSL:
![image](https://user-images.githubusercontent.com/50956430/194049005-d877bba0-ad1f-4cd0-bf1c-1cc4aaa8df3e.png)
RGB variant:
![image](https://user-images.githubusercontent.com/50956430/194049285-7b356b46-2796-4fdb-97b2-27bfbd223f9c.png)

Personally I liked first variant more.